src/ORM/Marshaller.php

```diff
@@ -216,7 +216,8 @@ protected function _prepareDataAndOptions($data, $options)
 
         $tableName = $this->_table->alias();
         if (isset($data[$tableName])) {
-            $data = $data[$tableName];
+            $data += $data[$tableName];
+            unset($data[$tableName]);
```

Comment:
I didn't notice that `unset` was needed... Does associated data affect the save process? Then this change would introduce another issue.

```php
[
    'Articles' => [
        'id' => 1,     
        'Comments' => [
            // Before this change, this record was available.
        ]
    ],
    'Comments' => [
        // After this change, this record will be available.
    ],
]
```